internal/controllers/topology/cluster/cluster_controller_test.go

```diff
@@ -791,6 +863,8 @@ func setupTestEnvForIntegrationTests(ns *corev1.Namespace) (func() error, error)
 		infrastructureClusterTemplate2,
 		infrastructureMachineTemplate1,
 		infrastructureMachineTemplate2,
+		infrastructureMachinePoolTemplate1,
+		infrastructureMachinePoolTemplate2,
```

Comment:
We also have to add the secrets for the clusterclass tracker.

We can create them above the initObjs:

```go
       cluster1Secret := kubeconfig.GenerateSecret(cluster1, kubeconfig.FromEnvTestConfig(env.Config, cluster1))
       cluster2Secret := kubeconfig.GenerateSecret(cluster2, kubeconfig.FromEnvTestConfig(env.Config, cluster2))
       // Unset the ownerrefs otherwise they are invalid because they contain an empty uid.
       cluster1Secret.ObjectMeta.OwnerReferences = nil
       cluster2Secret.ObjectMeta.OwnerReferences = nil
```

And add `cluster1Secret` and `cluster2Secret` to the `initObjs` array.